jetty-http2/http2-common/src/main/java/org/eclipse/jetty/http2/IStream.java

```diff
@@ -119,6 +119,13 @@
      */
     boolean isRemotelyClosed();
 
+    /**
+     * Fail all data queued in the stream and reset
+     * demand to 0.
+     * @param x the exception to fail the data with.
+     */
+    void failAllData(Throwable x);
```

Comment:
We could directly call the channel to tell it to fail the content instead of going through `HttpInput` to do that, but we would still have to call `HttpInput` for two reasons:

1) `HttpInput`'s `ContentProducer` may contain a content object that also needs to be failed.
2) `HttpInput` needs to know if EOF was reached after all content was failed to eventually abort the channel if there is some unconsumed input still in transit.

I'm on the fence between concentrating all that logic in `HttpInput.consumeAll()` that calls back into the channel or letting each channel take care of failing its content then asking `HttpInput` to fails its content too passing it a flag to say if EOF was detected by the channel or not. Both have readability drawbacks IMHO.